core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -206,10 +207,22 @@ public ReadBuilder rename(String fullName, String newName) {
       return this;
     }
 
+    /**
+     * Field name to field Id map, constructed from the table properties
+     * The current mapping is from name to field id. Aliases of a column
+     * are stored as separate entries in the map with the same value
+     * TODO: Address case insenstivity
+     * TODO: could a schema have duplicate column names?
+     */
+    public ReadBuilder extFieldNameToId(Map<String, Integer> extFieldNameToId) {
```

Comment:
The assumption I'm making is the the field names in the map are not canonical. This has the problem of not being able to store duplicate names in the schema. Let me know if we should use canonical names here.